src/ts/relayforwarder.ts

```diff
@@ -0,0 +1,96 @@
+import { keccak256, arrayify, defaultAbiCoder } from "ethers/utils";
+import { Wallet } from "ethers/wallet";
+import { ReplayProtectionAuthority } from "./replayprotectionauthority";
+import { RelayHub, ChainID } from "..";
+import {
+  ForwardParams,
+  DeploymentParams,
+  RelayCallData,
+  Forwarder,
+} from "./forwarder";
+
+/**
+ * A single library for approving meta-transactions and its associated
+ * replay protection. All contracts must support the msgSender() standard.
+ */
+export class RelayHubForwarder extends Forwarder<RelayCallData> {
+  /**
+   * Sets up the RelayHub Forwarder that relies on the msgSender() standard.
+   * It can only be used for a single wallet.
+   * @param chainID MAINNET or ROPSTEN
+   * @param relayHub RelayHub
+   * @param signer Signer's wallet
+   * @param replayProtectionAuthority Extends implementation ReplayProtectionAuthority
+   */
+  constructor(
+    chainID: ChainID,
+    relayHub: RelayHub,
```

Comment:
Again, probably best just to pass in an address. In fact I think that's all you actually use in this class.

Reply:
Is this for the relayhub? I use it for encoding the forward function later on

Reply:
I think you should just do an attach inside the constructor, so that the caller doesnt have to do that.

Reply:
done